enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,549 @@
+before the API server. Therefore, we will have to manually add extra
+metadata to those pod definitions, instead of relying on the admission
+hook to do it.
+
+Some of the workloads that will be classified as management are
+outside of the OpenShift release payload. Therefore we will need to
+document this feature so that workload authors can use it. We will
+also identify add-on components that we know our initial customers
+will use and to which we contribute, and make the changes to add the
+labels ourselves, starting with
+
+* performance-addon-operator
+* sriov-network-operator
+* ptp-operator
+* local-storage-operator
+* cluster-logging (fluentd)
+* klusterlet (RHACM)
+
+### User Stories
+
+#### Radio Access Network (RAN) Use Case
+
+In the context of telcocommunications service providers' 5G Radio Access
+Networks, it is increasingly common to see "cloud native" implementations
+of the 5G Distributed Unit (DU) component. Due to latency constraints,
+this DU component needs to be deployed very close to the radio antenna for
+which it is responsible. In practice, this can mean running this
+component on anything from a single server at the base of a remote cell
+tower or in a datacenter-like environment serving several base stations.
+
+A hypothetical DU example is an unusually resource-intensive workload,
+requiring 20 dedicated cores, 24 GiB of RAM consumed as huge pages,
+multiple SR-IOV NICs carrying several Gbps of traffic each, and
+specialized accelerator devices. The node hosting this workload must
+run a realtime kernel, be carefully tuned to ensure low-latency
+requirements can be met, and be configured to support features like
+Precision Timing Protocol (PTP).
+
+The most constrained resource for RAN deployments is CPU, so for now
+the focus of this enhancement is managing CPU requirements.
+Kubernetes resource requests for CPU time are measured in fractional
+"[Kubernetes
+CPUs](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu)". For
+bare metal, 1 Kubernetes CPU is equivalent to 1 hyperthread.
+
+Due to the resource-intensive workload, the overhead for platform
+components such as the OpenShift platform and add-on operators is
+severely restricted when compared to other uses. For example, one
+customer has allocated 4 hyperthreads for all components other than
+their own workloads. OpenShift currently requires 7.
+
+### Implementation Details/Notes/Constraints
+
+#### High-level End-to-end Workflow
+
+This section outlines an end-to-end workflow for deploying a cluster
+with management-workload-partitioning enabled and how pods are
+correctly scheduled to run on the management CPU pool.
+
+1. User sits down at their computer.
+2. The user creates their `install-config.yaml`, including extra
+   values in the `bootstrapInPlace` section (since that enables
+   deploying a single-node).
+   * `managementCPUIDs` is a CPU set specifier for the CPUs to add to
```

Comment:
Before we approve this I kind of expect the answer about why this isn't just a specially named cpu pool

Reply:
IIRC, the CRI-O team didn't want to add a field to all pools listing the CPUs and we wanted to reserve the ability to use runtime classes for other purposes. Maybe I'm mixing up concepts, though? @mrunalp or @haircommander ?

Reply:
kubelet itself knows two pools:

shared / dedicated

all we are doing is potentially subdividing the shared pool, but its not exclusive.
i see no way to make it exclusive without confusing cpu manager.
i also prefer that we keep the number of potential pools closely aligned to system/kube-reservation.

maybe we have 

```
- system-cpu-pools
  - name: management
    cpuIDs: 0-1
  - name: networking
    cpuIDs: 2-3
``

but all of these named pools are a subset of the kubelet understood shared pool and must fit within the reservation for cpu on the node.

Reply:
issue with the above is if we have `workload.io.openshift=management,networking` as the type of label.

i would probably say enumerating it as a list, but we just support one name for now may be safest as potentially we may want to move log collection or networking on a different set in future?

Reply:
we would have a label cpu.workload.openshift.io/management like node role probably,

we need to make a case whether a workload classification is exclusive or inclusive.  i’m pretty sure we would have more labels if we needed them since something can be both management and another classification, but i’m not convinced we need it yet

Reply:
We can make everything generic up and down the stack so if needed we can support other pools in the future.